File: Code/SimpleBudget.API/Models/TaxSettings/TaxSettingGridModel.cs
namespace SimpleBudget.API.Models
{
    public class TaxSettingGridModel
    {
        public int Year { get; set; }
        public decimal? CPPRate { get; set; }
        public decimal? CPPMaxAmount { get; set; }
        public decimal? EIRate { get; set; }
        public decimal? EIMaxAmount { get; set; }
        public decimal? CPPBasicExemptionAmount { get; set; }
        public decimal? FederalBasicPersonalAmount { get; set; }
        public decimal? ProvincialBasicPersonalAmount { get; set; }
        public decimal? CanadaEmploymentBaseAmount { get; set; }
    }
}

File: Code/client/src/api/models/TaxSettingGridModel.ts
export interface TaxSettingGridModel {
    year: number;
    cppRate: number | undefined | null;
    cppMaxAmount: number | undefined | null;
    eiRate: number | undefined | null;
    eiMaxAmount: number | undefined | null;
    cppBasicExemptionAmount: number | undefined | null;
    federalBasicPersonalAmount: number | undefined | null;
    provincialBasicPersonalAmount: number | undefined | null;
    canadaEmploymentBaseAmount: number | undefined | null;
}
